types/nightwatch/index.d.ts

```diff
@@ -6683,7 +6683,10 @@ export interface WebDriverProtocolScreenCapture {
      * @example
      * browser.screenshot(true);
      */
-    screenshot(log_screenshot_data: boolean, callback?: (screenshotEncoded: string) => void): this;
+    screenshot(
+        log_screenshot_data: boolean,
+        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
+    ): Awaitable<this, string>;
```

Comment:
Actually, I looked at the code and found that `.screenshot` can be used in two ways, with and without `log_screenshot_data` parameter.

So, we can use the function overloading here to specify both ways:

```suggestion
    screenshot(
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
    ): Awaitable<this, string>;
    screenshot(
        log_screenshot_data: boolean,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
    ): Awaitable<this, string>;
```

Reply:
Please update the tests as well after making this change to include both the function overloads.

Reply:
Thanks @garg3133 . If `log_screenshot_data ` is optional, what about 
```suggestion
    screenshot(
        log_screenshot_data?: boolean,
        callback?: (this: NightwatchAPI, result: NightwatchCallbackResult<string>) => void,
    ): Awaitable<this, string>;
```
Is that providing the same as the function overloading but more simple?

Reply:
@garg3133 Thanks for the explanation. That makes sense to me.